lib/src/main/kotlin/com/swmansion/starknet/service/http/requests/HttpBatchRequest.kt

```diff
@@ -0,0 +1,65 @@
+package com.swmansion.starknet.service.http.requests
+
+import com.swmansion.starknet.provider.Request
+import com.swmansion.starknet.provider.rpc.JsonRpcRequest
+import com.swmansion.starknet.provider.rpc.buildJsonHttpBatchDeserializer
+import com.swmansion.starknet.provider.rpc.buildJsonHttpBatchDeserializerOfDifferentTypes
+import com.swmansion.starknet.service.http.HttpService
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import java.util.concurrent.CompletableFuture
+
+class HttpBatchRequest<T> private constructor(
+    private val payload: HttpService.Payload,
+    private val deserializer: HttpResponseDeserializer<List<Result<T>>>,
+    private val service: HttpService,
+) : Request<List<Result<T>>> {
+
+    constructor(
+        url: String,
+        jsonRpcRequests: List<JsonRpcRequest>,
+        responseDeserializers: List<KSerializer<T>>,
+        deserializationJson: Json,
+        service: HttpService,
+    ) : this(
+        HttpService.Payload(
+            url = url,
+            method = "POST",
+            params = emptyList(),
+            body = Json.encodeToString(jsonRpcRequests),
+        ),
+        deserializer = buildJsonHttpBatchDeserializer(responseDeserializers, deserializationJson),
+        service = service,
+    )
+
+    constructor(
+        url: String,
+        jsonRpcRequests: List<JsonRpcRequest>,
+        responseDeserializers: List<KSerializer<out T>>,
+        deserializationJson: Json,
+        service: HttpService,
+        //  The `containsDifferentRequestTypes` parameter is added to overcome a type erasure problem in JVM
+        //  avoiding a clash between similar constructor signatures with generic types.
+        //  Its value doesn't affect the logic.
+        containsDifferentRequestTypes: Boolean,
```

Comment:
@DelevoXDG please have a say on this. 
It's a bummer we cannot make do without this, but if this is the only way to have single class with multiple cnstructors for batches I guess it's fine. Maybe we could assign a default value to it?

Reply:
> It's a bummer we cannot make do without this

I don't think this should matter anyway, as we normally should only create create `HttpBatchRequest` using `JsonProvider` methods. We might as well make these constructors internal.

> Maybe we could assign a default value to it?

I don't think we can do that, this will lead to the same clash as before. Personally, I would rather use named factory methods instead of one or both of the constructors, similarly to how we deal with e.g. `Felt.fromSigned`. Imo this should be cleaner than using a flag can accept all kinds of values.

Reply:
Great idea, I added `fromRequests` and `fromRequestsAny` to `HttpBatchRequests`. This way those 2 constructors have been removed and the only one is now private (same as in `HttpRequest`).